Cargo.toml

```diff
@@ -30,6 +30,7 @@ logger = []
 docs = ["unstable"]
 sessions = ["async-session"]
 unstable = []
+testing = ["surf"]
```

Comment:
This is somewhat unfortunate; in order to write tests an opt-in feature needs to be enabled. Testing is a fairly common capability, and ideally we could have it always be on.

I understand that this is because we always compile `surf`; but for example #697 doesn't have this issue, and I think it's something we should do better.